azure-functions-java-sdktypes/src/main/java/com/microsoft/azure/functions/sdktype/blob/BlobContainerHydrator.java

```diff
@@ -0,0 +1,136 @@
+package com.microsoft.azure.functions.sdktype.blob;
+
+import com.microsoft.azure.functions.sdktype.SdkTypeHydrator;
+
+import java.lang.reflect.Method;
+import java.util.logging.Logger;
+
+/**
+ * Reflection logic for building a BlobContainerClient from BlobContainerMetaData,
+ * supporting both connection string usage and managed identity usage.
+ */
+public class BlobContainerHydrator implements SdkTypeHydrator<BlobContainerMetaData> {
+    private static final Logger LOGGER = Logger.getLogger(BlobContainerHydrator.class.getName());
+
+    @Override
+    public Object createInstance(BlobContainerMetaData metaData) throws Exception {
+        String containerName = metaData.getContainerName();
+        String envVar = metaData.getConnectionEnvVar();
+
+        LOGGER.info("BlobContainerHydrator: Using environment variable name as prefix: " + envVar);
+
+        // Check if environment variable => connection string
+        String maybeConnString = System.getenv(envVar);
+        if (maybeConnString != null && isConnectionString(maybeConnString)) {
+            LOGGER.info("Detected connection string usage from environment variable: " + envVar);
+            return buildWithConnectionString(containerName, maybeConnString);
+        } else {
+            // interpret envVar as prefix for managed identity
+            LOGGER.info("Detected managed identity usage. Prefix: " + envVar);
+
+            String accountName = System.getenv(envVar + "__accountName");
```

Comment:
Use your discretion but I personaly move all constant strings to a private static variables and closer to the class.